### PR TITLE
2054 Owasp scans copied to prod DAC

### LINKED ISSUE #2054: As TDP Sys Admin, I need nightly owasp scan artifacts copied to production backend
**Description:**
Since #1754 was merged into `HHS/TANF-app:master` successful nightly scan reports do not appear in DAC. There are currently no reports in prod. See evidence [here](https://github.com/raft-tech/TANF-app/issues/2054#issuecomment-1213474067). 

These reports are not being copied to production backend in DAC because the repository that these artifacts are saved in is hard-coded in [`process_owasp_scan.py`](https://github.com/raft-tech/TANF-app/blob/4852f108f79e5cbb0b954cc8abbeaa5483bc5170/tdrs-backend/tdpservice/security/management/commands/process_owasp_scan.py#L55). `project-slug` should = `HHS/TANF-app` for production (and staging). 

**Acceptance Criteria:**
- [ ] Successful production owasp nightly scan reports are saved in production backend (i.e. `https://api-tanfdata.acf.hhs.gov/admin/security/owaspzapscan/`) 
- [ ] Successful dev and staging nightly scan reports are saved in respective non-prod backends
- [ ] Errors are raised when artifacts cannot be copied to backend environment. 
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [ ] update `process_owasp_scan.py` (re: project slug , error-handling)
- [ ]  Run Testing Checklist and confirm all tests pass

**Notes:**
- [original implementation of nightly owasp scan storage](https://github.com/raft-tech/TANF-app/pull/1340)
- see investigation notes [here](https://github.com/raft-tech/TANF-app/issues/2054#issuecomment-1216985625)
- this problem also applies to staging deployments from `HHS:main`

### PR DESCRIPTION
**Pull request by n0remac**
## Summary of Changes
_Provide a brief summary of changes_

When the nightly owasp job runs a different project slug will be passed to the DAC upload script for staging vs prod.

Pull request closes #2054 

## How to Test
_List the steps to test the PR_
Log in to staging or develop and check the scans in the admin page.
https://tdp-backend-staging.app.cloud.gov/admin/security/owaspzapscan/
https://tdp-backend-develop.app.cloud.gov/admin/security/owaspzapscan/

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Successful production owasp nightly scan reports are saved in production backend (i.e. `https://api-tanfdata.acf.hhs.gov/admin/security/owaspzapscan/`) 
- [x] Successful dev and staging nightly scan reports are saved in respective non-prod backends
- [x] Errors are raised when artifacts cannot be copied to backend environment. 
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary
- [x] lfrohlich and/or adpennington confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

Are all areas of code introduced in this PR meaningfully tested?
 - [x]If this PR introduces backend code changes, are they meaningfully tested?
 - [x] If this PR introduces frontend code changes, are they meaningfully tested?
Are code coverage minimums met?
 - [x] Frontend coverage: [94.29% %] (see CodeCov Report comment in PR)
 - [x] Backend coverage: [95.20% %] (see CodeCov Report comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

_n/a: devops_ 

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?
_n/a: devops_ 

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [x] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented?

### CODE REVIEW

**Review by andrew-jameson — APPROVED** [review]
Code is clean and straight forward, let's see it run and fill out the testing steps for Alex :+1:

File: .circleci/config.yml [hunk 1]
```diff
@@ -516,6 +516,13 @@ jobs:
       - run:
           name: Run post-processing task to record OWASP ZAP results
           command: |
+            # The project slug is used to identify the project that CircleCI is runing for.
+            # It is found here by taking the git url and removing the prefix.
+            # For example:
+            #   https://github.com/raft-tech/TANF-app -> raft-tech/TANF-app
+            #   https://github.com/HHS/TANF-app -> HHS/TANF-app
+            PROJECT_SLUG=$(echo << pipeline.project.git_url >> | sed 's/https:\/\/github.com\///g')
```
Comment (n0remac) [inline .circleci/config.yml]: clean this up

File: .circleci/config.yml [hunk 2]
```diff
@@ -516,6 +516,9 @@ jobs:
       - run:
           name: Run post-processing task to record OWASP ZAP results
           command: |
+            # Construct the project slug from the current branch name and user
+            PROJECT_SLUG=$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME
```
Comment (n0remac) [inline .circleci/config.yml:520]: This form or syntax was tested here:
https://app.circleci.com/pipelines/github/raft-tech/TANF-app/10383/workflows/081df837-f61d-4a59-921b-bd4fb3740c04/jobs/26496

**Review by ADPennington — APPROVED** [review]
lgtm! thanks @n0remac 🚀

**Review by jtimpe — APPROVED** [review]

**Pull request merged**